tests/regtest.py

```diff
@@ -0,0 +1,2002 @@
+
+		{
+			"easy": ("/some/file user:group mode"),
+			"secure": ("/some/file other:group mode", "/other/file user:group mode"),
+			[...]
+		}
+		"""
+
+		for profile, lines in entries.items():
+
+			with open(self.getProfilePath(profile), 'a') as profile_file:
+				for line in lines:
+					profile_file.write(line + "\n")
+
+	def addPackageProfileEntries(self, package, entries):
+		"""Just like addProfileEntries, but for a specific package in
+		permissions.d."""
+
+		for profile, lines in entries.items():
+
+			with open(self.getPackageProfilePath(package, profile), 'a') as profile_file:
+				for line in lines:
+					profile_file.write(line + "\n")
+
+	def buildProfileLine(self, path, mode, owner = "root", group = "root", caps = []):
+		ret = "{} {}:{} {}".format(
+			path, owner, group,
+			# chkstat doesn't like the 0oNNN syntax, so remove the
+			# 'o'
+			oct(mode).replace('o', '')
+		)
+
+		if caps:
+			ret += "\n +capabilities {}".format( ','.join(caps) )
+
+		return ret
+
+	def createSecuritySysconfig(self, permissions_val, fscaps_val = None):
+		"""Creates a /etc/sysconfig/security configuration using the
+		given configuration values.
+
+		:param str permissions_val: the value for the PERMISSION_SECURITY
+		                            setting. e.g. "easy local"
+		:param str fscaps_val: the value for the PERMISSIONS_FSCAPS
+		                            setting. e.g. yes/no
+		"""
+
+		items = {
+			"PERMISSION_SECURITY": permissions_val,
+			"PERMISSION_FSCAPS": fscaps_val
+		}
+
+		with open(self.m_sysconfig_security, 'w') as sec_file:
+
+			for key, val in items.items():
+				if not val:
+					val = ""
+				elif '"' in val:
+					raise Exception("{} must not contain quotes".format(
+						key
+					))
+
+				sec_file.write("{}=\"{}\"\n".format(key, val))
+
+	def switchSystemProfile(self, profile):
+		print("Switching to", profile if profile else "(empty)", "system permissions profile")
+		# configure the given profile as default
+		self.createSecuritySysconfig(profile)
+
+	def applySystemProfile(self, extra_args = []):
+		print("Applying current system profile using chkstat")
+		args = ["--system"] + extra_args
+		return self.callChkstat(args)
+
+	def extractPerms(self, s):
+		return s.st_mode & ~(stat.S_IFMT(s.st_mode))
+
+	def printMode(self, path):
+		s = os.stat(path)
+		perms = self.extractPerms(s)
+
+		print("Current status of {path}: {owner}:{group} {mode}".format(
+			path = path,
+			owner = pwd.getpwuid(s.st_uid).pw_name,
+			group = grp.getgrgid(s.st_gid).gr_name,
+			mode = str(oct(perms)).replace('o', '')
+		))
+
+	def printError(self, *args, **kwargs):
+
+		color_printer.setRed()
+		print("FAILURE:", *args, **kwargs)
+		color_printer.reset()
+		self.m_errors += 1
+
+	def printWarning(self, *args, **kwargs):
+
+		color_printer.setYellow()
+		print("WARNING:", *args, **kwargs)
+		color_printer.reset()
+		self.m_warnings += 1
+
+	def getMode(self, path):
+		s = os.stat(path)
+		return self.extractPerms(s)
```

Comment:
Looking at the current code I don't see a big benefit in changing the API at the moment. It would rather make things more complex, even though those two tiny functions aren't exactly elegant. I'd rather keep it as is for now.

Reply:
Well, I still don't quite see the purpose of this function but I won't block the PR over it.